packages/bsky/src/hydration/actor.ts

```diff
@@ -146,18 +150,36 @@ export class ActorHydrator {
   async getProfileViewerStatesNaive(
     dids: string[],
     viewer: string,
+    hydrateKnownFollowers = false,
   ): Promise<ProfileViewerStates> {
     if (!dids.length) return new HydrationMap<ProfileViewerState>()
-    const res = await this.dataplane.getRelationships({
-      actorDid: viewer,
-      targetDids: dids,
-    })
+
+    const subjectDidToKnownFollowersMap = {}
+    const knownFollowersRequests = hydrateKnownFollowers
```

Comment:
Could we do this without mutating an outside map? And instead resolve an array of values, then use the index in the reduce function afterwards to grab the relevant known follows?

Reply:
hmm yknow could also be nice to just move all this logic out into a separate helper function on this class 🤔 

idk not required, but you could make a nice little hydration map that cleans up the reducer logic later on as well